functions/internal/util/util.go

```diff
@@ -141,6 +145,20 @@ var (
 	notFoundError = NewBadRequestError(errors.New("not found"))
 )
 
+// NewStatusUnsupportedProtocolError wraps err in a StatusError whose HTTPStatusCode method
+// returns http.StatusTeapot and whose Message method returns
+// "unsupported protocol HTTP" to avoid responding to HTTP requests
+func NewStatusUnsupportedProtocolError(err error) StatusError {
```

Comment:
I'd prefer to just go with this generic function; we don't expect anyone outside of this package to need this functionality (and the name `NewStatusUnsupportedProtocolError` is misleading since we're actually sending 418). Also, with respect to the error, we just need to construct an error that we actually intend for the user to see. E.g., we could call this as `newStatusError(http.StatusTeapot, errors.New("unsupported protocol HTTP; only HTTPS is supported"))`.

```go
// newStatusError constructs a new statusError with the given code and error.
// The given error will be used as the message returned by StatusError.Message.
func newStatusError(code int, err error) statusError {
    return statusError {
        code: code,
		error: err,
		// Leave empty so that error.Error() will be used as the return value
		// from Message.
        message: "",
    }
}
```